doc/en/backwards-compatibility.rst

```diff
@@ -0,0 +1,12 @@
+.. _backwards-compatibility:
+
+Backwards Compatibility Policy
+==============================
+
+Keeping backwards compatibility has a very high priority in the pytest project. Although we have deprecated functionality over the years, most of it is still supported. All deprecations in pytest were done because simpler or more efficient ways of accomplishing the same tasks have emerged, making the old way of doing things unnecessary.
+
+With the pytest 3.0 release we introduced a clear communication scheme for when we will actually remove the old busted joint and politely ask you to use the new hotness instead, while giving you enough time to adjust your tests or raise concerns if there are valid reasons to keep deprecated functionality around.
```

Comment:
while this sentence is fun to read, my *personal* oppinion is that we should pick a more conservative way to say it for the documentation

my first idea for a more moring way to say it would be `we will actually remove aged and outdated features to open a path for future developments`